build.sh

```diff
@@ -18,14 +18,33 @@ if [ $# -lt 1 ]; then
   echo "- linux-armv6"
   echo "- linux-armv7"
   echo "- linux-arm64v8"
+  echo "- darwin-x64"
   echo
   exit 1
 fi
 VERSION_VIPS="$1"
 PLATFORM="${2:-all}"
 
+# macOS
+# Note: we intentionally don't build these binaries inside a Docker container
+if [ $PLATFORM = "darwin-x64" ]; then
+  export VERSION_VIPS
+  export PLATFORM
+  export RUST_TARGET="x86_64-apple-darwin"
+
+  # Added -fno-stack-check to workaround a stack misalignment bug on macOS 10.15
+  # See:
+  # https://forums.developer.apple.com/thread/121887
+  # https://trac.ffmpeg.org/ticket/8073#comment:12
```

Comment:
Does this mean that AVX is enabled for the macOS build? If so, do we know how? We may need to ensure more recent intrinsics such as AVX512 are not enabled to ensure support for older hardware.

Reply:
It's a leftover from my early attempt to compile with Homebrew's LLVM toolchain. `llvm-strip` failed for some archive files.
https://github.com/kleisauke/libvips-packaging/runs/688284884?check_suite_focus=true#step:7:6375

In the end, adding `-fno-stack-check` didn't solve this, but switching to the Clang provided by Xcode did. I think it's a safe change to leave this in. FFmpeg seems to do the same by default for macOS 10.15, see:
http://git.videolan.org/?p=ffmpeg.git;a=commitdiff;h=07ecc4615c3ef5cc2fc61209a845cffd0c073b13

If you want, I could do a test release without this compiler flag to make sure if it's not causing any side-effects.

Regarding AVX; as fair as I know it's only conditionally enabled on runtime within libjpeg-turbo, see:
https://github.com/libjpeg-turbo/libjpeg-turbo/blob/6d8caa9f88184ad1698e91c94b37acb9506dacf0/simd/x86_64/jsimd.c#L39